src/main/java/com/flowingcode/vaadin/addons/localecombobox/LocaleComboBox.java

```diff
@@ -0,0 +1,79 @@
+/*-
+ * #%L
+ * LocaleComboBox Add-on
+ * %%
+ * Copyright (C) 2024 Flowing Code
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package com.flowingcode.vaadin.addons.localecombobox;
+
+import com.vaadin.flow.component.combobox.ComboBox;
+import com.vaadin.flow.component.dependency.CssImport;
+import com.vaadin.flow.component.dependency.NpmPackage;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.data.renderer.LitRenderer;
+import java.util.Arrays;
+import java.util.Locale;
+
+/**
+ * Vaadin ComboBox extension that allows to choose between multiple locales.
+ * 
+ * @author Tomas Peiretti / Flowing Code
+ */
+@SuppressWarnings("serial")
+@NpmPackage(value = "flag-icons", version = "7.2.3")
+@CssImport("flag-icons/css/flag-icons.min.css")
+@CssImport("styles/locale-combo-box.css")
+public class LocaleComboBox extends ComboBox<Locale> {
+
+  private static final String ITEM_LAYOUT_CLASS_NAME = "fc-locale-combo-box-item-layout";
+  private static final String ITEM_FLAG_CLASS_NAME = "fc-locale-combo-box-item-flag";
+
+  /**
+   * Creates a new instance of LocaleComboBox with all the installed locales.
+   */
+  public LocaleComboBox() {
+    setItemLabelGenerator(Locale::getDisplayName);
+    setRenderer(getLocaleRenderer());
+    addValueChangeListener(this::onValueChange);
+    setItems(
+        Arrays.stream(Locale.getAvailableLocales()).filter(loc -> loc.getCountry().length() == 2)
+            .sorted((l1, l2) -> l1.getDisplayName().compareTo(l2.getDisplayName())).toList());
+  }
+
+  private LitRenderer<Locale> getLocaleRenderer() {
+    return LitRenderer
+        .<Locale>of("<vaadin-horizontal-layout class=\"" + ITEM_LAYOUT_CLASS_NAME + "\">"
```

Comment:
Since the minimum Java version is 17, you can use multiline text blocks, which don't require escaping `"`:

```
of("""
"a" "b"
""");